src/TemperatureZero.cpp

```diff
@@ -7,31 +7,135 @@
+}
+
+#define INT1V_DIVIDER_1000                1000.0
+#define ADC_12BIT_FULL_SCALE_VALUE_FLOAT  4095.0
+
+// Get all factory calibration parameters and process them
+// This includes both the temperature sensor calibration as well as the 1v reference calibration
+void TemperatureZero::getFactoryCalibration() {
+   // Factory room temperature readings
+  uint8_t roomInteger = (*(uint32_t*)FUSES_ROOM_TEMP_VAL_INT_ADDR & FUSES_ROOM_TEMP_VAL_INT_Msk) >> FUSES_ROOM_TEMP_VAL_INT_Pos;
+  uint8_t roomDecimal = (*(uint32_t*)FUSES_ROOM_TEMP_VAL_DEC_ADDR & FUSES_ROOM_TEMP_VAL_DEC_Msk) >> FUSES_ROOM_TEMP_VAL_DEC_Pos;
+  _roomTemperature = roomInteger + convert_dec_to_frac(roomDecimal);
+  _roomReading = ((*(uint32_t*)FUSES_ROOM_ADC_VAL_ADDR & FUSES_ROOM_ADC_VAL_Msk) >> FUSES_ROOM_ADC_VAL_Pos);
+   // Factory hot temperature readings
+  uint8_t hotInteger = (*(uint32_t*)FUSES_HOT_TEMP_VAL_INT_ADDR & FUSES_HOT_TEMP_VAL_INT_Msk) >> FUSES_HOT_TEMP_VAL_INT_Pos;
+  uint8_t hotDecimal = (*(uint32_t*)FUSES_HOT_TEMP_VAL_DEC_ADDR & FUSES_HOT_TEMP_VAL_DEC_Msk) >> FUSES_HOT_TEMP_VAL_DEC_Pos;
+  _hotTemperature = hotInteger + convert_dec_to_frac(hotDecimal);
+  _hotReading = ((*(uint32_t*)FUSES_HOT_ADC_VAL_ADDR & FUSES_HOT_ADC_VAL_Msk) >> FUSES_HOT_ADC_VAL_Pos);
+  // Factory internal 1V voltage reference readings at both room and hot temperatures
+  int8_t roomInt1vRefRaw = (int8_t)((*(uint32_t*)FUSES_ROOM_INT1V_VAL_ADDR & FUSES_ROOM_INT1V_VAL_Msk) >> FUSES_ROOM_INT1V_VAL_Pos);
+  int8_t hotInt1vRefRaw  = (int8_t)((*(uint32_t*)FUSES_HOT_INT1V_VAL_ADDR & FUSES_HOT_INT1V_VAL_Msk) >> FUSES_HOT_INT1V_VAL_Pos);
+  _roomInt1vRef = 1 - ((float)roomInt1vRefRaw/INT1V_DIVIDER_1000);
+  _hotInt1vRef  = 1 - ((float)hotInt1vRefRaw/INT1V_DIVIDER_1000);
+  // Combining the temperature dependent 1v reference with the ADC readings
+  _roomVoltageCompensated = ((float)_roomReading * _roomInt1vRef)/ADC_12BIT_FULL_SCALE_VALUE_FLOAT;
+  _hotVoltageCompensated = ((float)_hotReading * _hotInt1vRef)/ADC_12BIT_FULL_SCALE_VALUE_FLOAT;
+#ifdef TZ_DEBUG
+    Serial.println(F("\n+++ Factory calibration parameters:"));
+    Serial.print(F("Room Temperature : "));
+    Serial.println(_roomTemperature, 1);
+    Serial.print(F("Hot Temperature  : "));
+    Serial.println(_hotTemperature, 1);
+    Serial.print(F("Room Reading     : "));
+    Serial.println(_roomReading);
+    Serial.print(F("Hot Reading      : "));
+    Serial.println(_hotReading);
+    Serial.print(F("Room Voltage ref raw / interpreted : "));
+    Serial.print(roomInt1vRefRaw);
+    Serial.print(F(" / "));
+    Serial.println(_roomInt1vRef, 4);
+    Serial.print(F("Hot Voltage ref raw / interpreted  : "));
+    Serial.print(hotInt1vRefRaw);
+    Serial.print(F(" / "));
+    Serial.println(_hotInt1vRef, 4);
+    Serial.print(F("Room Reading compensated : "));
+    Serial.println(_roomVoltageCompensated, 4);
+    Serial.print(F("Hot Reading compensated  : "));
+    Serial.println(_hotVoltageCompensated, 4);
+#endif
+}
```

Comment:
@wsnel debug via serial it is not the best way to do it, we recommend removing or reviewing the following article https://www.sparkfun.com/news/3245 "Don’t Serial.print in your library"

Reply:
Great link, changes are in. Additionally examples are now more according to SparkFun recommendations as well.